html/inc/boinc_db.inc

```diff
@@ -95,7 +96,7 @@ class BoincDb extends DbConn {
     //
     static function get($dbnum = 0) {
         global $generating_xml;
-        if (isset(self::$instance)) {
+        if (isset(self::$instance) && intval(self::$dbnum) == intval($dbnum)) {
```

Comment:
The usual comment: changing `$dbnum` creates a new instance, leaving the old connection open until garbage collection kicks in. It would be better to actively close the connection as soon as it isn't needed anymore.

Reply:
Yes, but it's the duty of the calling web code to decide how long a specific instance is needed. That's why I asked for a close function in #5691. We in our server status page need the second connection only for a single query, then go back to using the first instance. Should we only use one at. atime?

Reply:
Note, #5691 deals with the C-API. Here you should be able to use `self::$instance->db_conn->close()`, no?

> it's the duty of the calling web code to decide how long a specific instance is needed

I disagree in this case since you explicitly "clobber" the existing instance. Callers don't have a say in what's happening here. Unless...

> We do save what get() returns in a local variable and don't care about the global instance.

... this changes things. Ok.